build-image/Dockerfile

```diff
@@ -10,45 +10,48 @@ RUN apt-get install -y nodejs && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
 # and viceversa.
 RUN npm install -g postcss-cli@5.0.1 autoprefixer@9.8.5
 
-ENV HUGO_VERSION=v0.72.0
-RUN git clone https://github.com/gohugoio/hugo.git --branch ${HUGO_VERSION} --depth 1 && \
-	cd hugo  && go install --tags extended && cd ../ && \
-	rm -rf hugo/ && rm -rf /go/pkg /go/src
-ENV SHFMT_VERSION=3.1.0
-RUN curl -fsSLo shfmt https://github.com/mvdan/sh/releases/download/v${SHFMT_VERSION}/shfmt_v${SHFMT_VERSION}_linux_amd64 && \
-	echo "cb91ea08a075a2f96b5230f09b4e211b7c108b1c97603caceb48d117d2ac5508  shfmt" | sha256sum -c && \
+ENV HUGO_VERSION=0.72.0
+RUN GOARCH=$(go env GOARCH) && echo $GOARCH && \
+	if [ "$GOARCH"="amd64" ]; then \
+		URL=https://github.com/gohugoio/hugo/releases/download/v${HUGO_VERSION}/hugo_${HUGO_VERSION}_Linux-64bit.tar.gz; \
+	elif [ "$GOARCH"="arm64" ]; then \
+		URL=https://github.com/gohugoio/hugo/releases/download/v${HUGO_VERSION}/hugo_${HUGO_VERSION}_Linux-ARM64.tar.gz; \
+	fi && \
+	echo Downloading ${URL} && \
+	curl -s -L -o hugo.tar.gz "${URL}" && \
+	tar xzvf hugo.tar.gz -C /usr/bin && \
+	rm hugo.tar.gz && \
+	chmod +x /usr/bin/hugo
+
+ENV SHFMT_VERSION=3.2.4
+RUN curl -fsSLo shfmt https://github.com/mvdan/sh/releases/download/v${SHFMT_VERSION}/shfmt_v${SHFMT_VERSION}_linux_$(go env GOARCH) && \
 	chmod +x shfmt && \
 	mv shfmt /usr/bin
+
 RUN curl -sfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh| sh -s -- -b /usr/bin v1.27.0
-RUN GO111MODULE=on go get -tags netgo \
+
+RUN GO111MODULE=on go get \
 		github.com/client9/misspell/cmd/misspell@v0.3.4 \
 		github.com/golang/protobuf/protoc-gen-go@v1.3.1 \
 		github.com/gogo/protobuf/protoc-gen-gogoslick@v1.3.0 \
 		github.com/gogo/protobuf/gogoproto@v1.3.0 \
-		github.com/weaveworks/tools/cover@bdd647e92546027e12cdde3ae0714bb495e43013 && \
-	rm -rf /go/pkg /go/src
+		github.com/weaveworks/tools/cover@bdd647e92546027e12cdde3ae0714bb495e43013 \
+		github.com/fatih/faillint@v1.5.0 \
+		github.com/campoy/embedmd@v1.0.0 \
+	&& true
 
-ENV KUBEVAL_VERSION=0.15.0
-RUN curl -Ls https://github.com/instrumenta/kubeval/releases/download/${KUBEVAL_VERSION}/kubeval-linux-amd64.tar.gz -o /tmp/kubeval-linux-amd64.tar.gz && \
-	tar -xf /tmp/kubeval-linux-amd64.tar.gz -C /tmp && \
-	cp /tmp/kubeval /usr/local/bin && \
-	rm -f /tmp/kubeval*
+# Cannot get it to run together in above go get.
+RUN GO111MODULE=on go get  \
+		github.com/instrumenta/kubeval@v0.16.1 \
+	&& true
+
+RUN rm -rf /go/pkg /go/src
```

Comment:
Could you move it back at the end of `go get` concatenating with `&&`? Reason is that image is not smaller if done in a separate RUN command because of the layered filesystem.

Reply:
Good point that I haven't realized. It's back now. Additionally, we now also delete /root/.cache, which makes images bit smaller.